src/renderer/apis/message-popover.ts

```diff
@@ -0,0 +1,45 @@
+import { Channel, Message } from "discord-types/general";
+import { Logger } from "../modules/logger";
+import { ButtonItem, getButtonItem } from "../../types/coremods/message";
+
+const logger = Logger.api("MessagePopover");
+
+export class MessagePopoverAPI extends EventTarget {
+  public buttons = new Map<string, getButtonItem>();
+
+  public addButton(identifier: string, item: getButtonItem): void {
+    this.buttons.set(identifier, item);
+  }
+
+  public removeButton(identifier: string): void {
+    this.buttons.delete(identifier);
+  }
```

Comment:
Instead of using an identifier based system, make `buttons` a `Set`. The `addButton` function can return a function that can be called to remove it.